### DISCUSSION TITLE
Is there any npc which can automove instead of static ?

### DISCUSSION THREAD

**Discussion by ndkk** [flareteam/flare-game#942] [Q&A] [answered]
I have just played this game, it's fantasy!  Like it !

But I can not find a npc which can automove.

I don't know whether it exist or not .

If it existed, maybe someone know it , please tell me, tks!

if it not exist, maybe someone can implement this feature ? 

I think it will be  a good feature! :)

**Comment by dorkster** [comment, marked as answer]
This should be possible now with this commit: https://github.com/flareteam/flare-engine/commit/13ca74aa6558ac4a7ed9110e6420f6c4a3e03f0d
Keep in mind, in order for an NPC to move at all, they'll need to have their `speed` and `chance_pursue` stats set to non-zero values.

**Reply by ndkk** [reply]
OK, tks! I'll try it !

**Reply by ndkk** [reply]
I had tried it, no problem, it worked well.  

Now I want another scene that when a enemy boss appears,  he can talk with me at first
 and then we can fight ? 

Can we implement this function ?

**Reply by CaptainFluffy1903** [reply]
> I had tried it, no problem, it worked well.
> 
> Now I want another scene that when a enemy boss appears, he can talk with me at first and then we can fight ?
> 
> Can we implement this function ?

I'm not sure but I think I see something like that at HERESY mod.

**Reply by CaptainFluffy1903** [reply]
> This should be possible now with this commit: [flareteam/flare-engine@13ca74a](https://github.com/flareteam/flare-engine/commit/13ca74aa6558ac4a7ed9110e6420f6c4a3e03f0d) Keep in mind, in order for an NPC to move at all, they'll need to have their `speed` and `chance_pursue` stats set to non-zero values.

Does this new non-static npcs can attack?

**Reply by ndkk** [reply]
no, it can't attack. I had tried yesterday.

but something seemed wield, the npc always wanted to avoid meeting me.

maybe it fears me？ ^ & ^

**Reply by CaptainFluffy1903** [reply]
> no, it can't attack. I had tried yesterday.
> 
> but something seemed wield, the npc always wanted to avoid meeting me.
> 
> maybe it fears me？ ^ & ^

Did u define dmg stats and a attack animation?

**Reply by ndkk** [reply]
I just define "[stance]" in animations/npcs/programmer1.txt.

as below: 

`image=images/npcs/programmer_left.png

[stance]
frames=4
duration=3s
type=looped
frame=0,0,0,0,48,64,24,56
frame=0,1,0,0,48,64,24,56
frame=0,2,0,0,48,64,24,56
frame=0,3,0,0,48,64,24,56
frame=0,4,0,0,48,64,24,56
frame=0,5,0,0,48,64,24,56
frame=0,6,0,0,48,64,24,56
frame=0,7,0,0,48,64,24,56
frame=1,0,48,0,48,64,24,56
frame=1,1,48,0,48,64,24,56
frame=1,2,48,0,48,64,24,56
frame=1,3,48,0,48,64,24,56
frame=1,4,48,0,48,64,24,56
frame=1,5,48,0,48,64,24,56
frame=1,6,48,0,48,64,24,56
frame=1,7,48,0,48,64,24,56
frame=2,0,96,0,48,64,24,56
frame=2,1,96,0,48,64,24,56
frame=2,2,96,0,48,64,24,56
frame=2,3,96,0,48,64,24,56
frame=2,4,96,0,48,64,24,56
frame=2,5,96,0,48,64,24,56
frame=2,6,96,0,48,64,24,56
frame=2,7,96,0,48,64,24,56
frame=3,0,144,0,48,64,24,56
frame=3,1,144,0,48,64,24,56
frame=3,2,144,0,48,64,24,56
frame=3,3,144,0,48,64,24,56
frame=3,4,144,0,48,64,24,56
frame=3,5,144,0,48,64,24,56
frame=3,6,144,0,48,64,24,56
frame=3,7,144,0,48,64,24,56
`
